src/mmw/js/src/modeling/models.js

```diff
@@ -888,6 +889,7 @@ var ScenarioModel = Backbone.Model.extend({
 var ScenariosCollection = Backbone.Collection.extend({
     model: ScenarioModel,
     comparator: 'created_at',
+    alerts: [],
```

Comment:
So the assumption is that since this is called from within `ScenarioCollection.updateScenarioName`, it could be called my multiple `ScenarioModel`s in quick succession, leading to unmanaged overlapping modals?

This will never happen. The only reason `updateScenarioName` is on the collection rather than on the model itself is we want to keep scenario names unique, and since a model has no reference to the collection to which it belongs we set it on the collection.

And since nothing from `this` is passed to the alert modal (only a string, and a warning config), we can simply use the pattern of creating an `modalViews.AlertView` and calling `render()` on it inline.